aiidalab_widgets_base/elns.py

```diff
@@ -0,0 +1,361 @@
+
+    return None
+
+
+class AppIcon:
+    def __init__(self, icon, link, description):
+        self.icon = icon
+        self.link = link
+        self.description = description
+
+    def to_html_string(self):
+        return f"""
+            <table style="border-collapse:separate;border-spacing:15px;">
+            <tr>
+                <td style="width:200px"> <a href="{self.link}" target="_blank">  <img src="{self.icon}"> </a></td>
+                <td style="width:800px"> <p style="font-size:16px;">{self.description} </p></td>
+            </tr>
+            </table>
+            """
+
+
+class DisplayAiidaObject(ipw.VBox):
+    node = traitlets.Instance(Node, allow_none=True)
+
+    def __init__(self, **kwargs):
+        self._output = ipw.Output()
+        super().__init__(
+            children=[
+                self._output,
+            ],
+            **kwargs,
+        )
+
+    @traitlets.observe("node")
+    def _observe_node(self, change):
+        from aiidalab_widgets_base import viewer
+
+        if change["new"] != change["old"]:
+            with self._output:
+                clear_output()
+                if change["new"]:
+                    display(viewer(change["new"]))
+
+
+class OpenInApp(ipw.VBox):
+
+    node = traitlets.Instance(Node, allow_none=True)
+
+    def __init__(self, **kwargs):
+        self.tab = ipw.Tab()
+        self.tab_selection = ipw.RadioButtons(
+            options=[],
+            description="",
+            disabled=False,
+            style={"description_width": "initial"},
+            layout=ipw.Layout(width="auto"),
+        )
+        spacer = ipw.HTML("""<p style="margin-bottom:1cm;"></p>""")
+        super().__init__(children=[self.tab_selection, spacer, self.tab], **kwargs)
+
+    @traitlets.observe("node")
+    def _observe_node(self, change):
+        if change["new"]:
+            self.tab.children = [
+                self.get_geo_opt_tab(),
+                self.get_geometry_analysis_tab(),
+                self.get_isotherm_tab(),
+            ]
+            self.tab.set_title(0, "Geometry Optimization")
+            self.tab.set_title(1, "Geometry analysis")
+            self.tab.set_title(2, "Isotherm")
+
+            self.tab_selection.options = [
+                (
+                    "Geometry Optimization - typically this is the first step needed to find optimal positions of atoms in the unit cell.",
+                    0,
+                ),
+                (
+                    "Geometry analysis - calculate parameters describing the geometry of a porous material.",
+                    1,
+                ),
+                (
+                    "Isotherm - compute adsorption isotherm of a small molecules in the selected material. ",
+                    2,
+                ),
+            ]
+            ipw.link((self.tab, "selected_index"), (self.tab_selection, "value"))
+        else:
+            self.tab.children = []
+
+    def get_geo_opt_tab(self):
```

Comment:
shall the notebooks take the `UUID` of the `StructureData` as url parameter and then users won't have to do this manual step and have the structure directly in the optimization app? Same probably applies for the other notebooks.

Reply:
this is already the case.